Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -160,5 +163,273 @@ public void RespectAnnotationPrefixes()
+        {
+            var serialized = SerializeToIonWithCustomSerializer(new NegativeDoubleIonSerializer(), 3.14);
+            Assert.AreEqual(-3.14, serialized.DoubleValue);
+        }
+
+        [TestMethod]
+        public void DeserializesWithCustomDoubleSerializer()
+        {
+            var deserialized = DeserializeWithCustomSerializer(new NegativeDoubleIonSerializer(), 3.14);
+            Assert.AreEqual(-3.14, deserialized);
+        }
+
+        [TestMethod]
+        public void SerializesWithCustomDecimalSerializer()
+        {
+            var serialized = SerializeToIonWithCustomSerializer(new NegativeDecimalIonSerializer(), (decimal)3.14);
+            Assert.AreEqual(-(decimal)3.14, serialized.DecimalValue);
+        }
+
+        [TestMethod]
+        public void DeserializesWithCustomDecimalSerializer()
+        {
+            var deserialized = DeserializeWithCustomSerializer(new NegativeDecimalIonSerializer(), (decimal)3.14);
+            Assert.AreEqual(-(decimal)3.14, deserialized);
+        }
+
+        [TestMethod]
+        public void SerializesWithCustomBigDecimalSerializer()
+        {
+            var testBigDecimal = BigDecimal.Parse("3.14159265359");
+            
+            var serialized = SerializeToIonWithCustomSerializer(new NegativeBigDecimalIonSerializer(), testBigDecimal);
+            Assert.AreEqual(-testBigDecimal, serialized.BigDecimalValue);
+        }
+
+        [TestMethod]
+        public void DeserializesWithCustomBigDecimalSerializer()
+        {
+            var testBigDecimal = BigDecimal.Parse("3.14159265359");
+            
+            var deserialized = DeserializeWithCustomSerializer(new NegativeBigDecimalIonSerializer(), testBigDecimal);
+            Assert.AreEqual(-testBigDecimal, deserialized);
+        }
+
+        [TestMethod]
+        public void SerializesWithCustomSymbolSerializer()
+        {
+            var serialized = SerializeToIonWithCustomSerializer(
+                new UpperCaseSymbolIonSerializer(), 
+                new SymbolToken("test symbol", 10));
+            Assert.AreEqual("TEST SYMBOL", serialized.SymbolValue.Text);
+        }
+
+        [TestMethod]
+        public void DeserializesWithCustomSymbolSerializer()
+        {
+            var deserialized = DeserializeWithCustomSerializer(
+                new UpperCaseSymbolIonSerializer(), 
+                new SymbolToken("test symbol", 10));
+            Assert.AreEqual("TEST SYMBOL", deserialized.Text);
+        }
+
+        [TestMethod]
+        public void SerializesWithCustomDateSerializer()
+        {
+            var testDate = new DateTime(2005, 05, 23);
+            var expectedDate = testDate.AddDays(1);
+
+            var serialized = SerializeToIonWithCustomSerializer(new NextDayDateTimeIonSerializer(), testDate);
+            Assert.AreEqual(expectedDate, serialized.TimestampValue.DateTimeValue);
+        }
+
+        [TestMethod]
+        public void DeserializesWithCustomDateSerializer()
```

Comment:
What about time and timestamp tests?

Reply:
`Timestamp` is an Ion type not a .NET type and these custom serializers are for .NET types (and `DateTime` is the .NET type for time/timestamps).